### PR TITLE
Rails 8, handle streaming, custom cdn, url

### PR DESCRIPTION
**Pull request by bendangelo**

### CODE REVIEW

**Comment by rkwap** [comment]
@bendangelo Thanks for raising the PR and making lot of changes. Apologies for the delay, and due to my busy schedule this week, I will be closing it this weekend.

**Comment by bendangelo** [comment]
No problem. Take a look at the other pr I opened as well.

File: lib/active_storage/service/bunny_service.rb [hunk 1]
```diff
@@ -8,13 +8,16 @@ module ActiveStorage
   # See ActiveStorage::Service for the generic API documentation that applies to all services.
   class Service::BunnyService < Service
 
-    CDN_BASE_URL = "https://#{ENV['BUNNY_STORAGE_ZONE']}.b-cdn.net"
-
     attr_reader :client
```
Comment (rkwap) [inline lib/active_storage/service/bunny_service.rb:11]: make base_url as a reader method to keep consistency with `ActiveStorage::Service` abstract class or any implemented class like `ActiveStorage::Service::S3Service`

File: lib/active_storage/service/bunny_service.rb [hunk 2]
```diff
@@ -25,10 +28,16 @@ def upload(key, io, checksum: nil, filename: nil, content_type: nil, disposition
...
-        raise ActiveStorage::FileNotFoundError
+      if block_given?
+        instrument :streaming_download, { key: key } do
+          stream(key, &block)
+        end
+      else
+        instrument :download, key: key do
+          io = StringIO.new object_for(key).get_file
+
+          io
```
Comment (rkwap) [inline lib/active_storage/service/bunny_service.rb:39]:
```suggestion
          io.set_encoding(Encoding::BINARY)
```

File: lib/active_storage/service/bunny_service.rb [hunk 3]
```diff
@@ -78,7 +97,7 @@ def private_url(key, expires_in:, filename:, disposition:, content_type:, **)
     end
 
     def public_url(key)
-      File.join(CDN_BASE_URL, key)
+      File.join(@base_url, key)
```
Comment (rkwap) [inline lib/active_storage/service/bunny_service.rb:100]:
```suggestion
      File.join(base_url, key)
```

File: lib/active_storage/service/bunny_service.rb (hunk 2 above)
Comment (rkwap) [reply lib/active_storage/service/bunny_service.rb:39]: better to move this to another private method since it is being used again in stream.

**Review by rkwap — CHANGES_REQUESTED** [review]
Minor refactoring. Rest LGTM!

**Comment by rkwap** [comment]
Do test out after making changes and leave a comment. Will merge ASAP when ready.

**Comment by bendangelo** [comment]
Done

**Comment by rkwap** [comment]
Thank you for your contribution :)
merging!

**Pull request merged**